operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/rate/LimitingRateOverPeriod.java

```diff
@@ -0,0 +1,21 @@
+package io.javaoperatorsdk.operator.processing.event.rate;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
+public @interface LimitingRateOverPeriod {
```

Comment:
Would rather have it names `RateLimited` or `RateLimitedOverPeriod`, but such rate limiting is always over period, so maybe just the simplest possible.